nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ReplaceText.java

```diff
@@ -673,6 +689,63 @@ public boolean isAllDataBufferedForEntireText() {
         }
     }
 
+    /**
+     * Replaces the first occurrence of a search value within a flow file with the given replacement value. It can
+     * either be done line by line, meaning that the first occurrence within a line gets replaced, or for the entire
+     * text, where only the very first occurrence is replaced.
+     */
+    private class ReplaceFirst implements ReplacementStrategyExecutor {
+        @Override
+        public FlowFile replace(FlowFile flowFile, ProcessSession session, ProcessContext context, String evaluateMode,
+                Charset charset, int maxBufferSize) {
+            final String replacementValue = context.getProperty(REPLACEMENT_VALUE)
+                    .evaluateAttributeExpressions(flowFile).getValue();
+            final AttributeValueDecorator quotedAttributeDecorator = Pattern::quote;
```

Comment:
According to the java documentation, you should quote the replacement value as well.